app/javascript/src/NewService/components/FormTemplate.jsx

```diff
@@ -0,0 +1,103 @@
+        id: 'service_name'
+      }
+    }
+  }
+}
+
+const renderLabel = (namespace: {htmlFor: string, label: string}, required: boolean = false) => <label htmlFor={namespace.htmlFor}>{namespace.label}
+  { required && <abbr title="required">*</abbr> }
+</label>
+
+const renderSelectOptions = (options: string[]) => options.map(option => <option key={option} value={option}>{option}</option>)
+
+const renderSelect = (namespace: {name: string, id: string}, props: FormTemplateProps) => <select required="required" name={namespace.name} id={namespace.id} onChange={props.handleChangeNamespaces}
+  data-selected-value={namespace.id === 'service_namespace' ? props.selectedNamespace : ''}>
+  {props.namespaces && renderSelectOptions(props.namespaces)}
+</select>
+
+const ErrorMessage = ({error}) => <p style={{color: 'red'}}>{ error.errorMsg && `Sorry, your request has failed with the error: ${error.errorMsg}`}</p>
+
+const renderServiceDiscoveriItems = (props, serviceDiscoveryProps) => {
+  return (
+    <React.Fragment>
+      { props.error && <ErrorMessage error={props.error}/>}
+      <li id="service_name_input" className="string required">
+        { renderLabel(serviceDiscoveryProps.serviceDiscoverySelect.namespace, false) }
+        { renderSelect(serviceDiscoveryProps.serviceDiscoverySelect.namespace, props) }
+      </li>
+      <li>
+        { renderLabel(serviceDiscoveryProps.serviceDiscoverySelect.projects, false) }
+        { renderSelect(serviceDiscoveryProps.serviceDiscoverySelect.projects, props) }
+      </li>
+    </React.Fragment>
+  )
+}
+
+const renderNewServiceItems = () => {
+  return (
+    <React.Fragment>
+      <li id="service_name_input" className="string required">
+        {renderLabel({htmlFor: 'service_name', label: 'Name'}, true)}
+        <input maxLength="255" id="service_name" type="text" name="service[name]" autoFocus />
+      </li>
+      <li id="service_system_name_input" className="string required">
+        {renderLabel({htmlFor: 'service_system_name', label: 'System name'}, true)}
+        <input maxLength="255" id="service_system_name" type="text" name="service[system_name]" />
+        <p className="inline-hints">Only ASCII letters, numbers, dashes and underscores are allowed.</p>{/* TODO: Get this from rails */}
+      </li>
+      <li id="service_description_input" className="text optional">
+        {renderLabel({htmlFor: 'service_description', label: 'Description'}, false)}
+        <textarea rows="3" id="service_description" name="service[description]">
+        </textarea>
+      </li>
+    </React.Fragment>
+  )
+}
+
+const FormTemplate = (props: FormTemplateProps) => {
+  const {isFormDiscovery, isVisible, formActionPath} = props
+  const serviceDiscoveryProps = getServiceDiscoveryProps()
+
+  return (
+    <form className={`formtastic service ${isVisible ? '' : 'is-hidden'}`} id={isFormDiscovery ? 'service_discovery' : 'new_service'} action={formActionPath} acceptCharset="UTF-8" method="post">
```

Comment:
It should be `accept-charset` attribute

Reply:
@hallelujah in React attributes are camelCased, but it will render the standard accept-charset html style